src/main/java/com/sphy/hotelmanagementapplication/service/HotelService.java

```diff
@@ -105,35 +145,59 @@ public HotelDTO updateHotel(HotelDTO hotelDTO, List<RoomDTO> roomDTOS) throws Nu
 			existingHotel.setAreaName(hotelDTO.getAreaName());
 			Optional<Admin>  admin = adminRepository.findById(hotelDTO.getId());
 			admin.ifPresent(existingHotel::setOwner);
-			for (RoomDTO roomDTO : roomDTOS){
-				roomService.updateRoom(roomDTO);
-			}
+
 			hotelRepository.save(existingHotel);
```

Comment:
We need to return the modified entity here, the save method returns the object that was saved, if the save operation wa successful. Please get the Hotel object that was saved, convert it to DTO and return that, and not the initial HotelDTO